fields/types/name/test/type.js

```diff
@@ -0,0 +1,499 @@
+var NameType = require('../NameType');
+
+exports.initList = function (List) {
+	List.add({
+		name: NameType,
+	});
+};
+
+exports.testFieldType = function (List) {
+	describe('updateItem', function () {
+
+	});
+
+	describe('validateInput', function () {
+		it('should validate string input', function (done) {
+			List.fields.name.validateInput({
+				name: 'Max',
+			}, function (result) {
+				demand(result).be(true);
+				done();
+			});
+		});
+
+		it('should validate emtpy string input', function (done) {
+			List.fields.name.validateInput({
+				name: '',
+			}, function (result) {
+				demand(result).be(true);
+				done();
+			});
+		});
+
+		it('should validate undefined input', function (done) {
+			List.fields.name.validateInput({}, function (result) {
+				demand(result).be(true);
+				done();
+			});
+		});
+
+		it('should validate null input', function (done) {
+			List.fields.name.validateInput({
+				name: null,
+			}, function (result) {
+				demand(result).be(true);
+				done();
+			});
+		});
+
+		it('should invalidate numeric input', function (done) {
+			List.fields.name.validateInput({
+				name: 1,
+			}, function (result) {
+				demand(result).be(false);
+				done();
+			});
+		});
+
+		it('should invalidate object input', function (done) {
+			List.fields.name.validateInput({
+				name: { things: 'stuff' },
+			}, function (result) {
+				demand(result).be(false);
+				done();
+			});
+		});
+
+		it('should invalidate array input', function (done) {
+			List.fields.name.validateInput({
+				name: [1, 2, 3],
+			}, function (result) {
+				demand(result).be(false);
+				done();
+			});
+		});
+
+		it('should invalidate Boolean input', function (done) {
+			List.fields.name.validateInput({
+				name: true,
+			}, function (result) {
+				demand(result).be(false);
+				done();
+			});
+		});
+
+		it('should invalidate function input', function (done) {
+			List.fields.name.validateInput({
+				name: function () {},
+			}, function (result) {
+				demand(result).be(false);
+				done();
+			});
+		});
+
+		it('should invalidate regexp input', function (done) {
+			List.fields.name.validateInput({
+				name: /foo/,
+			}, function (result) {
+				demand(result).be(false);
+				done();
+			});
+		});
+
+		it('should invalidate date input', function (done) {
+			List.fields.name.validateInput({
+				name: Date.now(),
+			}, function (result) {
+				demand(result).be(false);
+				done();
+			});
+		});
+
+		describe('first name', function () {
+			it('should validate string input', function (done) {
+				List.fields.name.validateInput({
+					name_first: 'Max',
+				}, function (result) {
+					demand(result).be(true);
+					done();
+				});
+			});
+
+			it('should validate dot notation', function (done) {
+				List.fields.name.validateInput({
+					'name.first': 'Max',
+				}, function (result) {
+					demand(result).be(true);
+					done();
+				});
+			});
+
+			it('should validate emtpy string input', function (done) {
```

Comment:
Also need to check `name: { first: '' }` here and with all other places the check would be relevant